scipy/optimize/_hessian_update_strategy.py

```diff
@@ -188,15 +188,51 @@ def update(self, delta_x, delta_grad):
             return
         if self.first_iteration:
             # Get user specific scale
-            if self.init_scale == "auto":
+            if isinstance(self.init_scale, str) and self.init_scale == "auto":
                 scale = self._auto_scale(delta_x, delta_grad)
             else:
-                scale = float(self.init_scale)
-            # Scale initial matrix with ``scale * np.eye(n)``
+                scale = self.init_scale
+
+            # Check for complex: numpy will silently cast a complex array to
+            # a real one but not so for scalar as it raises a TypeError.
+            # Checking here brings a consistent behavior.
+            replace = False
+            if np.size(scale) == 1:
+                # to account for the legacy behavior having the exact same cast
+                scale = float(scale)
+            elif np.iscomplexobj(scale):
+                raise TypeError("init_scale contains complex elements, "
+                                "must be real.")
+            else:  # test explicitly for allowed shapes and values
+                replace = True
+                scale = np.array(scale)  # copy, will replace the original
```

Comment:
Better to use the copy command to make sure that the data is not still not owned

Reply:
`copy` defaults to true for `np.array` (not for `np.asarray`, but I've added it explicitly.